Tests/Fluent.Tests/Sql/SqlTests.cs

```diff
@@ -72,6 +73,323 @@ public void CanOperateSqlFromRollUpClient()
+                    .WithRegion(Region.USSouthCentral)
+                    .WithNewResourceGroup(rgName)
+                    .WithAdministratorLogin(sqlServerAdminName)
+                    .WithAdministratorPassword(sqlServerAdminPassword)
+                    .DefineDatabase(databaseName)
+                        .FromSample(SampleName.AdventureWorksLT)
+                        .WithBasicEdition()
+                        .Attach()
+                    .Create();
+                var dbFromSample = sqlServer.Databases.Get(databaseName);
+                Assert.NotNull(dbFromSample);
+                Assert.Equal(DatabaseEditions.Basic, dbFromSample.Edition);
+
+                var serverAutomaticTuning = sqlServer.GetServerAutomaticTuning();
+                Assert.Equal(AutomaticTuningServerMode.Unspecified, serverAutomaticTuning.DesiredState);
+                Assert.Equal(AutomaticTuningServerMode.Unspecified, serverAutomaticTuning.ActualState);
+                Assert.Equal(4, serverAutomaticTuning.TuningOptions.Count);
+
+                serverAutomaticTuning.Update()
+                    .WithAutomaticTuningMode(AutomaticTuningServerMode.Auto)
+                    .WithAutomaticTuningOptions("createIndex", AutomaticTuningOptionModeDesired.Off)
+                    .WithAutomaticTuningOptions("dropIndex", AutomaticTuningOptionModeDesired.On)
+                    .WithAutomaticTuningOptions("forceLastGoodPlan", AutomaticTuningOptionModeDesired.Default)
+                    .Apply();
+                Assert.Equal(AutomaticTuningServerMode.Auto, serverAutomaticTuning.DesiredState);
+                Assert.Equal(AutomaticTuningServerMode.Auto, serverAutomaticTuning.ActualState);
+                Assert.Equal(AutomaticTuningOptionModeDesired.Off, serverAutomaticTuning.TuningOptions["createIndex"].DesiredState);
+                Assert.Equal(AutomaticTuningOptionModeActual.Off, serverAutomaticTuning.TuningOptions["createIndex"].ActualState);
+                Assert.Equal(AutomaticTuningOptionModeDesired.On, serverAutomaticTuning.TuningOptions["dropIndex"].DesiredState);
+                Assert.Equal(AutomaticTuningOptionModeActual.On, serverAutomaticTuning.TuningOptions["dropIndex"].ActualState);
+                Assert.Equal(AutomaticTuningOptionModeDesired.Default, serverAutomaticTuning.TuningOptions["forceLastGoodPlan"].DesiredState);
+
+                var databaseAutomaticTuning = dbFromSample.GetDatabaseAutomaticTuning();
+                Assert.Equal(4, databaseAutomaticTuning.TuningOptions.Count);
+
+                // The following results in "InternalServerError" at the moment
+                databaseAutomaticTuning.Update()
+                    .WithAutomaticTuningMode(AutomaticTuningMode.Auto)
+                    .WithAutomaticTuningOptions("createIndex", AutomaticTuningOptionModeDesired.Off)
+                    .WithAutomaticTuningOptions("dropIndex", AutomaticTuningOptionModeDesired.On)
```

Comment:
design thought: you are setting one option here right? shouldn't method name be in a single form - WithAutomaticTuningOption?

Reply:
The option name comes from a Dictionary which does not contain a static set of keys; it's a service returned map.

Reply:
sorry... missed the plural vs singular. Yes, it can be singular